homeassistant/components/group.py

```diff
@@ -249,40 +291,78 @@ def assumed_state(self):
 
     def update_tracked_entity_ids(self, entity_ids):
         """Update the member entity IDs."""
-        self.stop()
+        run_coroutine_threadsafe(
+            self.async_update_tracked_entity_ids(entity_ids), self.hass.loop
+        ).result()
+
+    @asyncio.coroutine
+    def async_update_tracked_entity_ids(self, entity_ids):
+        """Update the member entity IDs.
+
+        This method must be run in the event loop.
+        """
+        yield from self.async_stop()
         self.tracking = tuple(ent_id.lower() for ent_id in entity_ids)
         self.group_on, self.group_off = None, None
 
-        self.update_ha_state(True)
-
-        self.start()
+        yield from self.async_update_ha_state(True)
+        self.async_start()
 
     def start(self):
         """Start tracking members."""
-        self._unsub_state_changed = track_state_change(
-            self.hass, self.tracking, self._state_changed_listener)
+        run_callback_threadsafe(self.hass.loop, self.async_start).result()
+
+    def async_start(self):
+        """Start tracking members.
+
+        This method must be run in the event loop.
+        """
+        self._async_unsub_state_changed = async_track_state_change(
+            self.hass, self.tracking, self._state_changed_listener
+        )
 
     def stop(self):
         """Unregister the group from Home Assistant."""
-        self.remove()
+        run_coroutine_threadsafe(self.async_stop(), self.hass.loop).result()
+
+    @asyncio.coroutine
+    def async_stop(self):
+        """Unregister the group from Home Assistant.
+
+        This method must be run in the event loop.
+        """
+        self.async_remove()
```

Comment:
Should note, a test should've caught this, so there should likely be a test around this ensuring async_remove actually completes.